lib/strings.nix

```
@@ -253,7 +253,7 @@ rec {
       => false
   */
   hasInfix = infix: content:
-    builtins.match ".*${escapeRegex infix}.*" content != null;
+    builtins.match ".*${escapeRegex infix}.*" (toString content) != null;
```

that's probably a bit broad since toString will take just about anything. how about
```suggestion
    builtins.match ".*${escapeRegex infix}.*" (if builtins.isPath content or isDerivation content then toString content else content) != null;
```

---

... which is probably not broad enough because `__toString` exists. :facepalm: 
```suggestion
    let content' =
      if builtins.isPath content || isDerivation content || content ? __toString
      then toString content
      else content;
    in
      builtins.match ".*${escapeRegex infix}.*" content' != null;
```